accepted/0000-plug-an-play.md

```diff
@@ -0,0 +1,302 @@
+
+## 4. Solved Issues
+
+### A. INSTALL SPEED NOW REACHES NEW ALL-TIME-HIGHS
+
+Probably the most obvious win is that by keeping the dependencies files stored into the cache, Yarn doesn't have to copy them around anymore. This allows the link step to be skipped almost entirely - the last remaining actions being done only because of how Yarn is currently architectured and will be removed later on (we're still creating a deep tree before flattening it in a later step, which is unnecessary).
+
+### B. Installs can now be efficiently cached even on ci
+
+Now that files don't have to be copied anymore, efficient caching becomes a breeze. The actual Yarn cache can be persisted into locations shared between all CI instances, making it possible to skip installs altogether provided both the cache and the `.pnp.js` file are made available.
+
+Where some projects were spending more than two minutes running (like is the case for react-native, for example), Plug'n'Play now allows them to spend this time actually running their tests, decreasing the load on the CIs and making for a better developer experience - we all hate waiting for the tests to end.
+
+### C. users working on multiple projects across a system won't pay increasing install costs
+
+A common occurence in the Javascript world is developers working on multiple disconnected projects sharing similar dependencies (for example all projects created through `create-react-app`). Due to how package managers currently work, the files used by those projects were typically copied from the cache into multiple `node_modules`, multiplying both the total size of the installs on the disk and the time wasted running installs.
+
+Now that the files are read directly from the cache, no matter your system, you'll only ever pay the cost of having multiple projects once. This multi-megs project is much more bearable now that you know that its dependencies will be reused by all other projects on your machine now.
+
+### D. “perfect” hoisting due to the removal of the filesystem limitations
+
+An example will be worth a thousand words: let's say you have the following dependency tree:
+
+* `top-level`
+    * `package-a`
+        * `package-c@1.0.0`
+    * `package-b`
+        * `package-c@1.0.0`
+    * `package-c@2.0.0`
+
+In this situation, even though it is found multiple times inside the dependency tree, `package-c@1.0.0` cannot be hoisted in such a way that it can be installing only once. This is because doing so would conflict with the strict requirement of `package-c@2.0.0` declared by the top-level.
+
+Since Plug'n'Play flattens the dependency tree while still preserving the links between the nodes, the paths Node will get will be the same for any `package-c@1.0.0` inside the dependency tree, causing the package to be instantiated a single time.
+
+### A. users cannot require dependencies that aren't listed in their dependencies

```

Comment:
A. => E.

Reply:
tfw you reorder sections and forget to update the identifiers

Reply:
tfw you reorder sections and forget to update the identifiers